### PR TITLE
Feature convert to spa

### PR DESCRIPTION
**Pull request by petyaNikleva**

### CODE REVIEW

File: style/main.css [hunk 1]
```diff
@@ -91,47 +94,29 @@ nav {
...
 }
 
 /* -------- Advertisement --------- */
-
-/* .advertisement {
-    background-color: burlywood;
-} */
  
 .advertisement-card {
     width: 20vw;
```
Comment (NastyaSinkevich) [inline style/main.css]: I see that you are using different types of units, but for consistency and adaptive webapp it is better try to use one kind of units of measure (of course sometime you will need and px or/and  rem/em6 but vw, vh, rem, em, px use at the same time isn't a good practice)

File: router/router.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import { render } from "../../node_modules/lit-html/lit-html.js";
+import { newsFeedTemplate} from "../pages/newsFeed/newsFeedTemplate.js"
+import { loginTemplate } from "../pages/login/loginTemplate.js";
+import { registerTemplate } from "../pages/register/registerTemplate.js";
+import { userListTemplate } from "../pages/user-list/userListTemplate.js";
+import { navTemplate } from "../pages/nav/navTemplate.js";
+
+const routes = {
+    '/': { 
+        mainView : newsFeedTemplate,
```
Comment (NastyaSinkevich) [inline router/router.js:11]: Nice idea for mainView and navView!

File: router/router.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        const mainContainer = document.getElementsByClassName('main-content')[0];
+
+        render(navHtml, navContainer);
+        render(mainHtml, mainContainer);
+    }
+}
+
+
+//
+export const changePage = (page) => {
```
Comment (NastyaSinkevich) [inline router/router.js]: Is this function still being used through the app?

File: services/authService.js [hunk 4]
```diff
@@ -38,4 +38,19 @@ export function getAllUsers() {
     return users;
 }
 
+export function setUserforEdit(email) {
```
Comment (NastyaSinkevich) [inline services/authService.js]: I think, that authService could be a class and used as singleton pattern. 
Try to practice this.

File: services/authService.js [hunk 5]
```diff
@@ -38,4 +38,19 @@ export function getAllUsers() {
...
+    localStorage.setItem('userForEdit', email);
+}
+
+export function getUserforEdit() {
+    const email = localStorage.getItem('userForEdit');
+    const user = getUser(email);
```
Comment (NastyaSinkevich) [inline services/authService.js]: `return getUser(email)`

you write just this

Comment (petyaNikleva) [reply services/authService.js]: I've corrected this. Thanks.

File: router/router.js (hunk 3 above)
Comment (petyaNikleva) [reply router/router.js]: No, I've deleted it from the code.

File: style/main.css (hunk 1 above)
Comment (petyaNikleva) [reply style/main.css]: Yes, you are absolutely right. I'll refactor css files.

File: pages/edit/edit.css [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+  background-color: #ddd;
+  outline: none; 
+}
+
+hr {
+  border: 1px solid #f1f1f1;
+  margin-bottom: 25px;
+}
+
+.registerbtn {
```
Comment (NastyaSinkevich) [inline pages/edit/edit.css]: maybe better to name variables in general style `register-btn`

File: pages/edit/edit.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
+import { getUserforEdit, register } from '../../services/authService.js';
+import { checkInputValid } from '../../helpers/validations.js';
+
+export function userForEdit () {
```
Comment (NastyaSinkevich) [inline pages/edit/edit.js:5]: Why we need this func? 
why we can't use getUserforEdit()?

File: pages/edit/edit.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+export function updateHandler (e) {
+    e.preventDefault();
+    const form = (e.target).parentElement;
+    const inputs = form.querySelectorAll('.input-control input');
+
+    let hasError = false;
+    const arrInputElements = [...inputs];
+    arrInputElements.forEach(element => { 
+        checkInputValid(element);
+        let isError = element.closest('.input-control').classList.contains('error');
```
Comment (NastyaSinkevich) [inline pages/edit/edit.js]: could be const, not let

Comment (petyaNikleva) [reply pages/edit/edit.js]: You are right. I've changed it.

File: pages/edit/edit.js (hunk 7 above)
Comment (petyaNikleva) [reply pages/edit/edit.js:5]: I don't want to import authSevice in template files. I would like to keep the following logic/ structure for all files:
editTemplate file -> communicates with edit.js file -> communicates with authService.js

File: pages/edit/edit.css (hunk 6 above)
Comment (petyaNikleva) [reply pages/edit/edit.css]: Yes, I changed naming of register-btn, login-btn, edit-btn, delete-btn in Html and css files.

File: router/router.js (hunk 2 above)
Comment (petyaNikleva) [reply router/router.js:11]: Thanks. I also like it.

File: style/main.css (hunk 1 above)
Comment (petyaNikleva) [reply style/main.css]: I think that it is also ready. I've changed "vw" with "%" also in some places I converted "px" to "rem". I still have "px" for font-size, min-width, max-width, borders, but I suppose that it is fine. If you prefer I could change all "px" with "rem"?

**Pull request merged**